src/training/ui/menu.rs

```diff
@@ -70,6 +70,15 @@ unsafe fn render_submenu_page(app: &App, root_pane: &mut Pane) {
     let tab_selected = app.tab_selected();
     let tab = app.menu_items.get(tab_selected).unwrap();
 
+    let submenu_ids = app.menu_items
+        .values() // Get the MultiStatefulList for each tab
+        .flat_map(|multi_stateful_list| multi_stateful_list.lists.iter() // For each StatefulList in this MultiStatefulList
+            .flat_map(|sub_stateful_list| sub_stateful_list.items.iter() // For each submenu in this StatefulList
+                .map(|submenu| submenu.submenu_id) // Grab the ID
+            )
+        )
+        .collect::<Vec<&str>>();
+
```

Comment:
These comments were for initial understanding, I think the code speaks for itself otherwise afterwards, so we should perhaps remove the clutter.
```suggestion
    let submenu_ids = app.menu_items
        .values() // Get the MultiStatefulList for each tab
        .flat_map(|multi_stateful_list| multi_stateful_list.lists.iter()
            .flat_map(|sub_stateful_list| sub_stateful_list.items.iter()
                .map(|submenu| submenu.submenu_id)
            )
        )
        .collect::<Vec<&str>>();
```